domain/modelconfig/service/service.go

```diff
@@ -324,9 +390,10 @@ func (v *spaceValidator) HasSpace(ctx context.Context, spaceName string) (bool,
 // - Agent version is not being changed.
 // - CharmhubURL is not being changed.
 // - Network space exists.
-// - Secret backend exists.
+// - Secret backend is valid for the type of model.
 // - There is no changes to authorized keys.
 func (s *Service) updateModelConfigValidator(
+	modelType model.ModelType,
```

Comment:
modelType should be fetched within this func. It makes no sense to make the call site do the extra work when we can do it in here do what we say on the tin.

Reply:
This avoids doing 2 queries when 1 will do the job. The caller makes a call to st to get the model uuid and the model type in one query on the model table. The model type is used here and the uuid used by the caller in the next step. I really didn't want to make 2 queries to the model table.